internal/services/paloalto/next_generation_firewall_vhub_local_rulestack_resource.go

```diff
@@ -0,0 +1,301 @@
+}
+
+func (r NextGenerationFirewallVHubLocalRuleStackResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.NextGenerationFirewallName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"rule_stack_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: localrulestacks.ValidateLocalRulestackID,
+		},
+
+		"network_profile": schema.VHubNetworkProfileSchema(),
+
+		// Optional
+		"dns_settings": schema.DNSSettingsSchema(),
+
+		"destination_nat": schema.DestinationNATSchema(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r NextGenerationFirewallVHubLocalRuleStackResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"plan": schema.PlanSchema(),
+
+		"pan_etag": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r NextGenerationFirewallVHubLocalRuleStackResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 2 * time.Hour,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.PaloAlto.FirewallClient
+			localRuleStackClient := metadata.Client.PaloAlto.LocalRulestacksClient
+			loc := ""
+			var model NextGenerationFirewallVHubLocalRuleStackModel
+
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id := firewalls.NewFirewallID(metadata.Client.Account.SubscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			ruleStackID, err := localrulestacks.ParseLocalRulestackID(model.RuleStackId)
+			if err != nil {
+				return err
+			}
+
+			ruleStack, err := localRuleStackClient.Get(ctx, *ruleStackID)
+			if err != nil {
+				return fmt.Errorf("reading %s for %s: %+v", ruleStackID, id, err)
+			}
+
+			loc = location.Normalize(ruleStack.Model.Location)
+
+			firewall := firewalls.FirewallResource{
+				Location: loc,
+				Properties: firewalls.FirewallDeploymentProperties{
+					AssociatedRulestack: &firewalls.RulestackDetails{
+						ResourceId: pointer.To(ruleStackID.ID()),
+						Location:   pointer.To(loc),
+					},
+					DnsSettings: schema.ExpandDNSSettings(model.DNSSettings),
+					MarketplaceDetails: firewalls.MarketplaceDetails{
+						OfferId:     "pan_swfw_cloud_ngfw",
+						PublisherId: "paloaltonetworks",
+					},
+					NetworkProfile: schema.ExpandNetworkProfileVHub(model.NetworkProfile),
+					PlanData: firewalls.PlanData{
+						BillingCycle: firewalls.BillingCycleMONTHLY,
+						PlanId:       "panw-cloud-ngfw-payg",
```

Comment:
these might want to be user configurable from the start?

Reply:
These are the only possible values at this time afaik.